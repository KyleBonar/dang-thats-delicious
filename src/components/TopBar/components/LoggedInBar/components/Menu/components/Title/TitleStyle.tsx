import styled from "../../../../../../../../theme/styled-components";

export const StyledDiv = styled.div`
  padding: 12px 12px 0 12px;
  text-transform: uppercase;
  font-size: 14px;
  color: ${x => x.theme.black};
`;
StyledDiv.displayName = "div";
