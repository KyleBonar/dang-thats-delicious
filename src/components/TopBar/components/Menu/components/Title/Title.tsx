import * as React from "react";
import styled from "styled-components";

const StyledDiv = styled.div`
  padding: 12px;
  background-color: ${x => x.theme.primaryLightThemeColor};
  text-transform: uppercase;
  font-size: 14px;
`;

interface TitleProps {
  children: string;
}

const Title: React.SFC<TitleProps> = props => {
  return <StyledDiv>{props.children}</StyledDiv>;
};

export default Title;
