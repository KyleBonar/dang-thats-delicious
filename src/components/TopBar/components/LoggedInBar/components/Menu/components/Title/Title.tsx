import * as React from "react";

import { StyledDiv } from "./TitleStyle";

export interface TitleProps {
  children: string;
  className?: string;
}

const Title: React.FC<TitleProps> = props => {
  return <StyledDiv className={props.className}>{props.children}</StyledDiv>;
};

export default Title;
