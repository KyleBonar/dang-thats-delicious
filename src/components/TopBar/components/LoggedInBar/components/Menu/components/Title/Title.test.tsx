import * as React from "react";
import * as enzyme from "enzyme";

import Title, { TitleProps } from "./Title";
import {
  casual,
  ITERATION_SIZE
} from "../../../../../../../../utils/testUtils/testUtils";

const fakeTitle = (): TitleProps => ({
  children: casual.string,
  className: casual.random_element([undefined, casual.string])
});

const mockTitles = new Array(ITERATION_SIZE).fill(null).map(fakeTitle);

describe("<Title />", () => {
  it("renders", () => {
    mockTitles.forEach(mockTitle => {
      const wrapper = enzyme.shallow(
        <Title {...mockTitle}>{mockTitle.children}</Title>
      );

      expect(wrapper).toBeTruthy();
    });
  });
});
