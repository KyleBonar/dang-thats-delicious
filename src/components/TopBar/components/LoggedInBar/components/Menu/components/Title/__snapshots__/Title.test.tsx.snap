// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> matches snapshot 1`] = `
<div
  className="magnam fugiat error molestiae laborum est possimus"
>
  sequi facilis corporis sed veritatis quidem praesentium
</div>
`;

exports[`<Title /> matches snapshot 2`] = `
<div
  className="vitae occaecati officia iure nisi ut maiores"
>
  aut enim iure facilis ut explicabo iure
</div>
`;

exports[`<Title /> matches snapshot 3`] = `
<div>
  fuga reprehenderit soluta non omnis consequatur enim
</div>
`;

exports[`<Title /> matches snapshot 4`] = `
<div>
  omnis adipisci eius laborum eius aut illo
</div>
`;

exports[`<Title /> matches snapshot 5`] = `
<div
  className="vel provident dicta perferendis dolorum cupiditate illum"
>
  tempora aut sint aperiam occaecati illum animi
</div>
`;

exports[`<Title /> matches snapshot 6`] = `
<div>
  ut assumenda sint quia sed exercitationem omnis
</div>
`;

exports[`<Title /> matches snapshot 7`] = `
<div>
  adipisci mollitia iste itaque omnis debitis rem
</div>
`;

exports[`<Title /> matches snapshot 8`] = `
<div>
  eaque nisi cum ducimus nemo doloremque maiores
</div>
`;

exports[`<Title /> matches snapshot 9`] = `
<div>
  non itaque voluptas ea ducimus autem sit
</div>
`;

exports[`<Title /> matches snapshot 10`] = `
<div
  className="est doloremque explicabo eaque necessitatibus sit suscipit"
>
  quam nulla est unde sit ex expedita
</div>
`;

exports[`<Title /> matches snapshot 11`] = `
<div>
  necessitatibus maiores ullam quae corporis qui aut
</div>
`;

exports[`<Title /> matches snapshot 12`] = `
<div
  className="est rerum at sint dolores fugiat quis"
>
  voluptate in sunt occaecati nulla aut sunt
</div>
`;

exports[`<Title /> matches snapshot 13`] = `
<div>
  voluptas culpa et vero repellat eveniet assumenda
</div>
`;

exports[`<Title /> matches snapshot 14`] = `
<div
  className="qui dignissimos dolore qui voluptatem cumque nihil"
>
  consectetur soluta inventore laudantium dolor sint officiis
</div>
`;

exports[`<Title /> matches snapshot 15`] = `
<div>
  est unde rem similique et maxime quia
</div>
`;

exports[`<Title /> matches snapshot 16`] = `
<div>
  distinctio autem quam qui facere in ut
</div>
`;

exports[`<Title /> matches snapshot 17`] = `
<div
  className="voluptate sit saepe assumenda deserunt consequatur itaque"
>
  at modi occaecati illum laborum quos eum
</div>
`;

exports[`<Title /> matches snapshot 18`] = `
<div>
  in fugit in rerum libero odit aut
</div>
`;

exports[`<Title /> matches snapshot 19`] = `
<div
  className="voluptatibus libero aut autem ut ullam et"
>
  ut ducimus fugiat et et aut quidem
</div>
`;

exports[`<Title /> matches snapshot 20`] = `
<div
  className="minus deserunt nihil quasi minus quo sunt"
>
  sint fugiat maiores autem est esse quos
</div>
`;

exports[`<Title /> matches snapshot 21`] = `
<div>
  earum animi quis expedita et optio consequatur
</div>
`;

exports[`<Title /> matches snapshot 22`] = `
<div>
  provident eaque atque quos rerum nobis nam
</div>
`;

exports[`<Title /> matches snapshot 23`] = `
<div>
  occaecati et consequatur consectetur sunt molestiae cupiditate
</div>
`;

exports[`<Title /> matches snapshot 24`] = `
<div
  className="ab assumenda animi non quis quis nobis"
>
  ipsa consequuntur quidem dolorem quia consequatur omnis
</div>
`;

exports[`<Title /> matches snapshot 25`] = `
<div>
  aliquam molestiae occaecati a culpa facilis eligendi
</div>
`;

exports[`<Title /> matches snapshot 26`] = `
<div
  className="quia aut sint cumque sunt mollitia quibusdam"
>
  voluptate voluptatem est eius dolorem amet quia
</div>
`;

exports[`<Title /> matches snapshot 27`] = `
<div>
  eum repudiandae minima temporibus ducimus quia assumenda
</div>
`;

exports[`<Title /> matches snapshot 28`] = `
<div
  className="architecto ut corrupti voluptatem quia dolores dolor"
>
  voluptas rem repellendus nihil repudiandae aliquid ex
</div>
`;

exports[`<Title /> matches snapshot 29`] = `
<div
  className="accusantium vel enim sint voluptatem porro ut"
>
  sequi perferendis nostrum aliquid ea animi architecto
</div>
`;

exports[`<Title /> matches snapshot 30`] = `
<div>
  sed qui repellat assumenda et incidunt aut
</div>
`;

exports[`<Title /> matches snapshot 31`] = `
<div>
  non sit et ut voluptatem quia dolor
</div>
`;

exports[`<Title /> matches snapshot 32`] = `
<div
  className="et dicta odit vel quisquam qui harum"
>
  sed nobis excepturi saepe autem quis molestiae
</div>
`;
